File: sdk/python-sdk/concise/exceptions.py
"""
Exception classes for Concise SDK
"""


class ConciseError(Exception):
    """Base exception for Concise SDK"""
    pass


class AuthenticationError(ConciseError):
    """Raised when API key is invalid or missing"""
    pass


class APIError(ConciseError):
    """Raised when API returns an error"""

    def __init__(self, message: str, status_code: int = None):
        super().__init__(message)
        self.status_code = status_code


class RateLimitError(ConciseError):
    """Raised when rate limit is exceeded"""
    pass


class NetworkError(ConciseError):
    """Raised when network request fails"""
    pass

File: sdk/typescript-sdk/src/exceptions.ts
/**
 * Exception classes for Concise SDK
 */

export class ConciseError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'ConciseError';
  }
}

export class AuthenticationError extends ConciseError {
  constructor(message: string) {
    super(message);
    this.name = 'AuthenticationError';
  }
}

export class APIError extends ConciseError {
  statusCode?: number;

  constructor(message: string, statusCode?: number) {
    super(message);
    this.name = 'APIError';
    this.statusCode = statusCode;
  }
}

export class RateLimitError extends ConciseError {
  constructor(message: string) {
    super(message);
    this.name = 'RateLimitError';
  }
}

export class NetworkError extends ConciseError {
  constructor(message: string) {
    super(message);
    this.name = 'NetworkError';
  }
}
